src/renderers/shared/fiber/ReactFiberScheduler.js

```diff
@@ -1188,15 +1189,32 @@ module.exports = function<T, P, I, TI, PI, C, CX, PL>(
       if (capturedErrors === null) {
         capturedErrors = new Map();
       }
-      capturedErrors.set(boundary, {
-        componentName,
-        componentStack,
-        error,
-        errorBoundary: errorBoundaryFound ? boundary.stateNode : null,
-        errorBoundaryFound,
-        errorBoundaryName,
-        willRetry,
-      });
+
+      if (__DEV__) {
+        capturedErrors.set(boundary, {
+          componentName,
+          componentStack,
+          error,
+          errorBoundary: errorBoundaryFound ? boundary.stateNode : null,
+          errorBoundaryFound,
+          errorBoundaryName,
+          willRetry,
+          shouldIgnoreErrorMessage: error != null &&
+            typeof error.__reactShouldIgnoreErrorMessage === 'boolean' &&
+            error.__reactShouldIgnoreErrorMessage,
```

Comment:
Why `typeof error.__reactShouldIgnoreErrorMessage === 'boolean' && error.__reactShouldIgnoreErrorMessage` instead of `error.__reactShouldIgnoreErrorMessage === true` ?